### PR TITLE
rsyslog testing - test k8s, es tls, viaq formatting

### PR DESCRIPTION
**Pull request by richm**
This tests the ability of rsyslog to do kubernetes metadata
annotation, elasticsearch with client cert auth, and some basic
viaq data model formatting

This currently uses pre-release rsyslog built in fedora copr, but
that should move into more official repos soon.

### CODE REVIEW

File: hack/testing/rsyslog/viaq_formatting.conf [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    if strlen($!message) == 0 then {
+        if strlen($!MESSAGE) > 0 then {
+            set $!message = $!MESSAGE;
+        } else {
+            if strlen($!log) > 0 then {
+                set $!message = $!log;
+            }
+        }
+    }
+    unset $!MESSAGE;
```
Comment (portante) [inline hack/testing/rsyslog/viaq_formatting.conf]: Should we only unset `$!MESSAGE` when we assign it at line 164?  And likewise unset `$!log` at line 167?

Comment (richm) [reply hack/testing/rsyslog/viaq_formatting.conf]: Neither `MESSAGE` nor `log` are fields in the common data model.  I haven't implemented support for pipeline_metadata - I think these should go into the `original_raw_message` field.

Comment (richm) [reply hack/testing/rsyslog/viaq_formatting.conf]: @portante we still have a long way to go before rsyslog is a drop in replacement for fluentd - it wasn't my intention to do that in this PR - this PR is only intended to test the basic k8s metadata functionality for $LARGE_CUSTOMER, and as a side effect start the other es and formatting work

- parsing of the JSON embedded in the log/MESSAGE field and assign those to top level fields - in this PR the parsing is done with mmnormalize, but it cannot set those fields as top level fields - see http://lists.adiscon.net/pipermail/rsyslog/2018-March/045207.html
- the rest of fluent-plugin-viaq: stripping empty fields, adding pipeline metadata, eventrouter
- docker audit log parsing
- crio file support
- other parts of the pipeline as found in various bits of fluentd .conf files

File: hack/testing/rsyslog/playbook.yaml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+      for name in $( oc get -n logging sa aggregated-logging-fluentd -o jsonpath='{.secrets[*].name}' ) ; do
+        case $name in
+        *-fluentd-token-*) echo $name ; break ;;
+        esac
+      done
+    register: fluentdsecret
+    when: use_mmk8s | default(True)
+
+  - name: get fluentd token
+    shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: can we replace this with?
```yaml
  command: >
    oc extract -n logging {{ fluentdsecret.stdout }} --keys=token --to=-
```

`command` is safer than `shell`

File: hack/testing/rsyslog/playbook.yaml [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      oc get -n logging secret {{ fluentdsecret.stdout }} --template='{{ '{{' }}index .data "token"{{ '}}' }}' | base64 -d
+    register: fluentdtoken
+    when: use_mmk8s | default(True)
+
+  - name: get fluentd CA cert for k8s
+    shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: same ask here

File: hack/testing/rsyslog/playbook.yaml [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    - use_mmk8s | default(True)
+
+  - name: install template config files
+    template: src={{ item }}.j2 dest=/etc/rsyslog.d/viaq/{{ item }} mode=0400
+    with_items:
+    - mmk8s.conf
+    when: use_mmk8s | default(True)
+
+  - name: get fluentd CA cert for ES
+    shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: command: oc extract

File: hack/testing/rsyslog/playbook.yaml [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+      oc get -n logging secret logging-fluentd --template='{{ '{{' }}index .data "ca"{{ '}}' }}' | base64 -d
+    register: fluentdcaes
+
+  - name: get fluentd client cert for ES
+    shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: command: oc extract

File: hack/testing/rsyslog/playbook.yaml [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+      oc get -n logging secret logging-fluentd --template='{{ '{{' }}index .data "cert"{{ '}}' }}' | base64 -d
+    register: fluentdcert
+
+  - name: get fluentd client key for ES
+    shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: command: oc extract

File: hack/testing/rsyslog/playbook.yaml [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+      oc get -n logging secret logging-fluentd --template='{{ '{{' }}index .data "key"{{ '}}' }}' | base64 -d
+    register: fluentdkey
+
+  - name: get es ip addr
+    shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: is it possible rewrite this into a `command` task instead?

File: hack/testing/rsyslog/playbook.yaml [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    copy: content={{ fluentdcert.stdout }} dest=/etc/rsyslog.d/viaq/es-cert.pem mode=0400
+
+  - name: install ES client key file
+    copy: content={{ fluentdkey.stdout }} dest=/etc/rsyslog.d/viaq/es-key.pem mode=0400
+
+  - name: handle es-ops
+    when: openshift_logging_use_ops
+    block:
+    - name: get es-ops ip addr
+      shell: |
```
Comment (ewolinetz) [inline hack/testing/rsyslog/playbook.yaml]: `shell` => `command`

**Comment by ewolinetz** [comment]
LGTM, squash and we can push it along

**Pull request merged**